### DISCUSSION TITLE
Attaching headers in custom routes

### DISCUSSION THREAD

**Discussion by TrippleCCC** [pocketbase/pocketbase#2421] [Q&A]
I'm experimenting with writing a custom route that responds with a file. I'm trying to use the `echo.Context.Attachment` and `echo.Context.Inline`, however, the `Content-Disposition` header doesn't seem to be populated. I then tried to attach a random header to the response and it wasn't being populated either. Is there a workaround for this?

**Comment by ganigeorgiev** [comment]
Could you provide a code sample with your custom route?

**Reply by ganigeorgiev** [reply]
I've tested with the below example and the custom header is written to the response for me:

```go
package main

import (
	"log"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.GET("/test", func(c echo.Context) error {
			c.Response().Header().Add("x-my-custom-header", "test")

			return c.Attachment("test.txt", "test")
		})

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```
